concrete/src/User/Notification/UserNotificationEventHandler.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Concrete\Core\User\Notification;
+
+use Concrete\Core\Entity\Notification\UserDeactivatedNotification;
+use Concrete\Core\Notification\Type\Manager;
+use Concrete\Core\Notification\Type\UserDeactivatedType;
+use Concrete\Core\User\Event\DeactivateUser;
+
+/**
+ * This method is less of a "service" and more of a controller
+ * It deals with deactivated user events
+ */
+class UserNotificationEventHandler
+{
+
+    /**
+     * The notification manager we send notifications with
+     *
+     * @var \Concrete\Core\Notification\Type\Manager
+     */
+    protected $notificationManager;
+
+    public function __construct(Manager $manager)
+    {
+        $this->notificationManager = $manager;
+    }
+
+    /**
+     * Handle deactivated user events
+     *
+     * @param \Concrete\Core\User\Event\DeactivateUser $event
+     */
+    public function deactivated(DeactivateUser $event)
+    {
+        /** @var UserDeactivatedType $type */
+        $type = $this->notificationManager->driver(UserDeactivatedType::IDENTIFIER);
+        $notifier = $type->getNotifier();
+
+        if (method_exists($notifier, 'notify')) {
```

Comment:
What about adding the `notify` method to the `NotifierInterface`? I know that it'd be a BC break, but I don't think that someone willl implement a notifier that doesn't have a notify method...

Reply:
:-1: I agree with your reasoning, but I don't like the idea of making breaking changes outside of major versions. And if we do want to go forward with it, we should do it in another pull request at least.

Reply:
Without using an interface, you are not sure what are the parameters of a method, you only know it exists (it may require more arguments than the ones you pass, or arguments of specific types)...